walkthroughs/ecs-service-connectivity/yelb/deployments/platformdeployment/AWS/ECS/yelb-cloudformation-ECS-AppMesh-deployment.yaml

```diff
@@ -0,0 +1,631 @@
+                        Value: '1'
+                      - Name: APPMESH_VIRTUAL_NODE_NAME
+                        Value:
+                            Fn::Join:
+                                - ''
+                                -
+                                    - mesh/
+                                    - !Ref Mesh
+                                    - /virtualNode/
+                                    - 'yelb-ui'
+                                    
+                
+
+
+    CloudWatchLogsGroup:
+        Type: AWS::Logs::LogGroup
+        Properties: 
+            LogGroupName: !Ref AWS::StackName
+            RetentionInDays: 365  
+   
+    YelbECSTaskIamRole:
+        Type: AWS::IAM::Role
+        Properties:
+            Path: /
+            AssumeRolePolicyDocument: |
+                {
+                    "Statement": [{
+                        "Effect": "Allow",
+                        "Principal": { "Service": [ "ecs-tasks.amazonaws.com" ]},
+                        "Action": [ "sts:AssumeRole" ]
+                    }]
+                }
+            ManagedPolicyArns:
+                - arn:aws:iam::aws:policy/CloudWatchFullAccess
+                - arn:aws:iam::aws:policy/AWSXRayDaemonWriteAccess
+                - arn:aws:iam::aws:policy/AWSAppMeshEnvoyAccess
+#                - arn:aws:iam::aws:policy/AWSAppMeshPreviewEnvoyAccess
+
+    YelbECSTaskExecutionRole:
+        Type: AWS::IAM::Role
+        Properties:
+            AssumeRolePolicyDocument:
+                Statement:
+                - Effect: Allow
+                  Principal:
+                     Service: [ecs-tasks.amazonaws.com]
+                  Action: ['sts:AssumeRole']
+            Path: /
+            ManagedPolicyArns:
+                - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+                - arn:aws:iam::aws:policy/CloudWatchLogsFullAccess
+
+    YelbServiceDiscoveryNameSpace:
+        Type: AWS::ServiceDiscovery::PrivateDnsNamespace
+        Properties:
+            Description: "Service Discovery Namespace for Yelb"
+            Vpc: !Ref 'VPC'
+            Name: !Ref 'Domain'
+
+
+    YelbAppserverServiceDiscoveryEntry:
+        Type: AWS::ServiceDiscovery::Service
+        Properties:
+            Name: yelb-appserver
+            DnsConfig:
+                DnsRecords: 
+                    - Type: A
+                      TTL: "10"
+                NamespaceId: !Ref 'YelbServiceDiscoveryNameSpace'
+            HealthCheckCustomConfig:
+                FailureThreshold: '1'
+                
+
+    YelbUiServiceDiscoveryEntry:
+        Type: AWS::ServiceDiscovery::Service
+        Properties:
+            Name: yelb-ui
+            DnsConfig:
+                DnsRecords: 
+                    - Type: A
+                      TTL: "10"
+                NamespaceId: !Ref 'YelbServiceDiscoveryNameSpace'
+            HealthCheckCustomConfig:
+                FailureThreshold: '1'
```

Comment:
Minor: I would set higher failure thresholds which would be representative of real settings. Maybe 2 or 3 instead?

Reply:
Done